roseau/load_flow/io/dict.py

```diff
@@ -22,6 +27,29 @@
 logger = logging.getLogger(__name__)
 
 
+@functools.lru_cache(maxsize=1)
+def _get_system_info() -> JsonDict:
+    from roseau.load_flow import __version__  # circular import
+
+    return {
+        # RLF version (for compatibility checks)
+        # --------------------------------------
+        "version": __version__,
+        # System information (useful for debugging)
+        # -----------------------------------------
+        "os_name": os.name,  # posix, nt etc.
+        "platform": sys.platform,  # linux, win32, darwin etc.
+        "machine": platform.machine(),  # x86_64, amd64, i386 etc.
+        # Python environment (useful for debugging)
+        # -----------------------------------------
+        # Only packages that may affect the json data are included
+        "python_version": sys.version_info,
+        "numpy_version": version("numpy"),
+        "pint_version": version("pint"),
+        "shapely_version": version("shapely"),
+    }
```

Comment:
I expected something like `{"version": 1}`. My purpose is just to make a check when parsing the JSON in `network_from_json`. I also would like the version of the data to be independent of the version of the software. 

Here, you are implementing a `show_version` function just like in [Pandas](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.show_versions.html) for instance. Good idea, but in another PR.